src/components/shared/Fields/Input/HookForm/InputComponent.tsx

```diff
@@ -62,30 +63,41 @@ const HookFormInputComponent = ({
   name,
   onChange,
   onBlur,
+  valueAsNumber,
+  value,
   ...restInputProps
 }: HookFormInputComponentProps) => {
-  const { register } = useFormContext();
-  const {
-    onChange: hookFormOnChange,
-    onBlur: hookFormOnBlur,
-    ref,
-  } = register(name);
+  const { control } = useFormContext();
+  const { field } = useController({
```

Comment:
One of the things I changed related to the dialog forms is that the amounts we store in the form state are actually numbers instead of strings. I did this so that the types and validation we assign to number fields match the actual value (This wasn't as obvious of a problem in the Dapp because the typing was looser there).

This logic is something I extracted from [here](https://react-hook-form.com/advanced-usage#TransformandParse). The idea is that whenever the onBlur and onChange handlers are fired on the inputs/Cleavers, instead of passing the raw value string to the form state we transform them into numbers.

This should also answer the question below. 

Let me know what you think!

Reply:
I've decided to discard these changes as it seems like I have forgotten about javascript inability to properly deal with large numbers, however, if we decide that we still want to transform input values to numbers in the feature, I still have those changes stored in my stash